patches/0001-Add-systemcrypto-GOEXPERIMENT.patch

```diff
@@ -0,0 +1,282 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Davis Goodin <dagood@microsoft.com>
+Date: Wed, 31 May 2023 16:54:31 -0500
+Subject: [PATCH] Add systemcrypto GOEXPERIMENT
+
+Adds systemcrypto GOEXPERIMENT and modifies tag handling to make
+goexperiment.systemcrypto behave as an alias that enables the recommended
+backend for the target GOOS.
+
+The reverse is also true: goexperiment.systemcrypto can be used as an alias in a
+build constraint. It's satisfied when (and only when) the recommended backend
+for the target GOOS is enabled, such as goexperiment.opensslcrypto on Linux.
```

Comment:
@qmuntal and I talked about `when (and only when)` a bit offline--I leaned towards symmetry when I was implementing this and lost track of how that makes the tag less useful for constraints:

* "Any backend", regardless of what Microsoft Go recommends, is useful. In our own patches, this would simplify
  `//go:build boringcrypto || goexperiment.opensslcrypto || goexperiment.cngcrypto`. Other library/app maintainers may need conditions just like that.
* Could cause breaks: if we change our Windows backend recommendation from (say) CNG to BetterCNG in Go 1.32, someone using a `//go:build goexperiment.systemcrypto` constraint and `GOEXPERIMENT=cngcrypto` for compatibility/long-term-support reasons would be broken in Go 1.32 because cngcrypto would no longer enable goexperiment.systemcrypto, only bettercngcrypto would. This seems like a needless break.

A quirk here is that `boringcrypto` is actually statically linked rather than being system-provided... but I'm not sure it's worth splitting hairs, just mention in the experiment doc.

We can always create an additional synthetic tag to distinguish "any backend" from "Microsoft-Go-picked automatic backend", but I want to give it a shot with just the one tag name first, and see if it ends up being fine.

That said, this is a tweak, so we don't think it should block this PR.

---

Earlier, I had some concern about what "any backend" might imply:

	// It may seem like it's enough to satisfy "goexperiment.systemcrypto" when
	// "goexperiment.opensslcrypto" is specified, and omit the GOOS=Linux check.
	// However, systemcrypto is a recommendation by the Microsoft Go toolset,
	// and we want to be careful not to misinform the user.

The idea is if you set `GOOS=windows GOEXPERIMENT=opensslcrypto` and your Go file has `//build goexperiment.systemcrypto`, the fact that it's satsified *could* be construed to mean that we recommend it, because passing `goexperiment.systemcrypto` enables our recommendation. But... if we simply document the behavior, I don't see this being an issue.

Reply:
Done. After writing it out and changing the code, I don't see any issue with this.